### ISSUE TITLE
Error starting Docker when creating a machine

### ISSUE THREAD

**Issue by tamlyn** [docker/machine#4225]
When I try to create a new machine (this is DigitalOcean but same happens on AWS):

```
$ docker-machine create --driver digitalocean --digitalocean-access-token $DOTOKEN test-machine
Running pre-create checks...
Creating machine...
(test-machine) Creating SSH key...
(test-machine) Creating Digital Ocean droplet...
(test-machine) Waiting for IP address to be assigned to the Droplet...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
Provisioning with ubuntu(systemd)...
Installing Docker...
Copying certs to the local machine directory...
Copying certs to the remote machine...
Setting Docker configuration on the remote daemon...
Error creating machine: Error running provisioning: ssh command error:
command : sudo systemctl -f start docker
err     : exit status 1
output  : Job for docker.service failed because the control process exited with error code. See "systemctl status docker.service" and "journalctl -xe" for details.
```

So I try

```
$ docker-machine ssh test-machine systemctl status docker.service
● docker.service - Docker Application Container Engine
   Loaded: loaded (/lib/systemd/system/docker.service; enabled; vendor preset: enabled)
  Drop-In: /etc/systemd/system/docker.service.d
           └─10-machine.conf
   Active: inactive (dead) (Result: exit-code) since Mon 2017-08-14 18:46:31 UTC; 1min 10s ago
     Docs: https://docs.docker.com
  Process: 4880 ExecStart=/usr/bin/docker daemon -H tcp://0.0.0.0:2376 -H unix:///var/run/docker.sock --storage-driver aufs --tlsverify --tlscacert /etc/docker/ca.pem --tlscert /etc/docker/server.pem --tlskey /etc/docker/server-key.pem --label provider=digitalocean (code=exited, status=1/FAILURE)
 Main PID: 4880 (code=exited, status=1/FAILURE)

Aug 14 18:46:30 test-machine systemd[1]: docker.service: Main process exited, code=exited, status=1/FAILURE
Aug 14 18:46:30 test-machine systemd[1]: Failed to start Docker Application Container Engine.
Aug 14 18:46:30 test-machine systemd[1]: docker.service: Unit entered failed state.
Aug 14 18:46:30 test-machine systemd[1]: docker.service: Failed with result 'exit-code'.
Aug 14 18:46:31 test-machine systemd[1]: docker.service: Service hold-off time over, scheduling restart.
Aug 14 18:46:31 test-machine systemd[1]: Stopped Docker Application Container Engine.
Aug 14 18:46:31 test-machine systemd[1]: docker.service: Start request repeated too quickly.
Aug 14 18:46:31 test-machine systemd[1]: Failed to start Docker Application Container Engine.
exit status 3
```

But I don't really know what that means.

```
$ docker-machine version
docker-machine version 0.12.0, build 45c69ad
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.6
```

Any ideas?

**Comment by peteralt** [comment]
Same issue with rackspace

**Comment by peteralt** [comment]
So for rackspace, I was able to fix this by adding the 
`--engine-storage-driver` overlay2` flag to the command.

**Comment by emabota** [comment]
Hi! 
Facing the same issue. @peteralt fix didn't work for me. 
PLEASE any hint why this is happening, and how to solve it? 
Thanks

**Comment by emabota** [comment]
Well, I updated my docker-machine, from `docker-machine version 0.12.0, build 45c69ad` to `docker-machine version 0.12.2, build 9371605`. It's working fine now.

**Comment by tamlyn** [comment]
Looks like the issue was fixed in 0.12.1:
> Fixed an issue with the version comparison function that prevented machines created with Engine 17.06.0 from starting properly.
